base_layer/contacts/src/contacts_service/storage/database.rs

```diff
@@ -188,6 +191,23 @@ where T: ContactsBackend + 'static
 
         Ok(())
     }
+
+    pub fn confirm_message(
+        &self,
+        message_id: Vec<u8>,
+        delivery_confirmation: Option<u64>,
+        read_confirmation: Option<u64>,
+    ) -> Result<(), ContactsServiceStorageError> {
+        self.db
+            .write(WriteOperation::Upsert(Box::new(DbKeyValuePair::MessageConfirmations(
+                message_id,
+                delivery_confirmation
+                    .map(|d| NaiveDateTime::from_timestamp_opt(i64::try_from(d).unwrap_or(0), 0).unwrap()),
+                read_confirmation.map(|d| NaiveDateTime::from_timestamp_opt(i64::try_from(d).unwrap_or(0), 0).unwrap()),
```

Comment:
dont really like the unwraps here.

Reply:
That's reasonable. I've updated it with better handling.